addons/mail/static/tests/owl/component/followers_menu_tests.js

```diff
@@ -0,0 +1,534 @@
+        intercepts: {
+            do_action: function (event) {
+                assert.step('add_followers_action');
+                assert.strictEqual(
+                    event.data.action.context.default_res_model,
+                    'res.partner',
+                    "'The 'add channels' action should contain thread model in context'");
+                assert.ok(
+                    event.data.action.context.mail_invite_follower_channel_only,
+                    "The 'add channels' action should be restricted to channels only");
+                assert.strictEqual(
+                    event.data.action.context.default_res_id,
+                    100,
+                    "The 'add channels' action should contain thread id in context");
+                assert.strictEqual(
+                    event.data.action.res_model,
+                    'mail.wizard.invite',
+                    "The 'add channels' action should be a wizard invite of mail module");
+                assert.strictEqual(
+                    event.data.action.type,
+                    "ir.actions.act_window",
+                    "The 'add channels' action should be of type 'ir.actions.act_window'");
+            },
+        },
+    });
+    const threadLocalId = this.env.store.dispatch('_createThread', {
+        id: 100,
+        _model: 'res.partner',
+    });
+    await this.createFollowersMenuComponent(threadLocalId);
+    assert.strictEqual(
+        document.querySelectorAll('.o_FollowersMenu').length,
+        1,
+        "should have followers menu component"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_FollowersMenu_buttonFollowers').length,
+        1,
+        "should have followers button"
+    );
+
+    await document.querySelector('.o_FollowersMenu_buttonFollowers').click();
+    await afterNextRender();
+    assert.strictEqual(
+        document.querySelectorAll('.o_FollowersMenu_dropdown').length,
+        1,
+        "followers dropdown should be opened"
+    );
+    assert.strictEqual(
+        document.querySelectorAll('.o_FollowersMenu_addChannelsButton').length,
+        1,
+        "followers dropdown should contain a 'add channels' button"
+    );
+    await document.querySelector('.o_FollowersMenu_addChannelsButton').click();
+    await afterNextRender();
+    assert.strictEqual(
+        document.querySelectorAll('.o_FollowersMenu_dropdown').length,
+        0,
+        "followers dropdown should be closed after click on 'add channels'"
+    );
+    assert.verifySteps(
+        ['add_followers_action'],
+        "Click on 'add channels' should trigger the right action"
+    );
+});
+
+QUnit.test('add followers', async function (assert) {
+    assert.expect(10);
+    const followerIds = [];
+    await this.start({
+        intercepts: {
+            do_action: function (e) {
+                assert.step('action:open_view');
+                followerIds.push(1);
+                e.data.options.on_close();
```

Comment:
I would add more assertions to check that this do_action is really related to opening of desired view.

In fact, and that's for all our tests that intercept `do_action`, I would not assume that there's always a single `do_action` in the flow and it's always the correct one. In theory, there may be much more unrelated do_action that may happen during the test, a bit like `trigger_up` or `rpc`.